core/include/gnuradio-4.0/BlockTraits.hpp

```diff
@@ -391,11 +424,10 @@ concept processBulk_requires_ith_output_as_span
                                                         traits::block::stream_output_port_types<TDerived>>::template apply<std::tuple>
                            bad_outputs) {
               {
-                  []<std::size_t... InIdx, std::size_t... OutIdx>(std::index_sequence<InIdx...>,
-                                                                  std::index_sequence<OutIdx...>) -> decltype(d.processBulk(std::get<InIdx>(inputs)..., std::get<OutIdx>(outputs)...)) {
-                      return {};
-                  }(std::make_index_sequence<traits::block::stream_input_port_types<TDerived>::size>(), std::make_index_sequence<traits::block::stream_output_port_types<TDerived>::size>())
+                  detail::can_processBulk_invoke_test(d, inputs, outputs, std::make_index_sequence<stream_input_port_types<TDerived>::size>(),
+                                                      std::make_index_sequence<stream_output_port_types<TDerived>::size>())
               } -> std::same_as<work::Status>;
+              // TODO: Is this check redundant?
```

Comment:
I would expect that additional check to  make this assertion pass, but apparently this is not working:

``` c++
// ...
    gr::work::Status
    processBulk(gr::ConsumableSpan auto, auto &)
    requires(processVariant == ProcessBulkVariant::AUTO_SPAN)
    {
        // do some bulk-type processing
        return gr::work::Status::OK;
    }
// ...
static_assert(!gr::traits::block::processBulk_requires_ith_output_as_span<BlockSignaturesProcessBulkSpan<float, ProcessBulkVariant::AUTO_SPAN>, 0>); // this assertion fails, while imo it shouldn't.
```

I tried to also replace the lambda construct with `can_processBulk` as done above, but that also didn't fix it.

It's also not a complete given that auto should always use consumable spans, but if not I fear that very subtle code changes like commenting out a method that only exists in ConsumableSpan to completely change the semantics of the code.

All not that critical since we should encurage constraining the method with a concept anyway.

Reply:
I've conducted tests using `auto` as well.  In practical code, `ConsumableSpan` would be used for `auto`. However, for the context of this concept, verifying compatibility with `auto` seems unnecessary given that it can accommodate any type as a template parameter.  

I still find this particular check to be redundant, but I leave a the code with comment for the moment.